src/api/types.h

```diff
@@ -23,5 +23,27 @@ typedef struct CGameSettings {
   const char* seat_controllers[4];
   int num_controllers = 0;
 };
+
+typedef struct CObservedGameState {
+  int currentPlayer;
+  int turnNum;
+  int roundNum;
+  int riichiSticks;
+  int counters;
+  int lastCall;
+  int lastCaller;
+  bool concealedKan;
+  uint64_t seed;
+  CPiece pendingPiece;
+  int scores[4];
+  int points[4];  // Player points
+  bool hasRonned[4];
+  CPiece hands[4][14];
+  CPiece* players[4];  // Pointers to discard arrays for each player
```

Comment:
I would rename this to be more obvious that it's the discards per player. Also we could probably create a struct similar to the Hand class and use that instead?

Reply:
probably ideal?